frontend/src/components/EmployeeModal/EmployeeModal.tsx

```diff
@@ -132,57 +145,137 @@ const EmployeeModal = ({
     if (imageBase64 !== '') {
       uploadPhotoForEmployee(updatedEmployee.id, table, refresh, false);
     }
+    return updatedEmployee;
+  };
+
+  const createOrUpdateDriver = async (
+    driver: AdminData | DriverData,
+    isNewDriver = false
+  ) => {
+    if (isNewDriver) {
+      return await createNewEmployee(
+        driver,
+        '/api/drivers',
+        () => refreshDrivers(),
+        'Drivers'
+      );
+    } else {
+      return await updateExistingEmployee(
+        driver,
+        '/api/drivers',
+        () => refreshDrivers(),
+        'Drivers'
+      );
+    }
+  };
+
+  const createOrUpdateAdmin = async (admin: AdminData, isNewAdmin = false) => {
+    if (isNewAdmin) {
+      await createNewEmployee(
+        admin,
+        '/api/admins',
+        () => refreshAdmins(),
+        'Admins'
+      );
+    } else {
+      await updateExistingEmployee(
+        admin,
+        '/api/admins',
+        () => refreshAdmins(),
+        'Admins'
+      );
+    }
+  };
+
+  const deleteDriver = async (id: string | undefined) => {
+    await axios.delete(`/api/drivers/${id}`);
+  };
+
+  const deleteAdmin = async (id: string | undefined) => {
+    await axios.delete(`/api/admins/${id}`);
   };
 
   const onSubmit = async (data: ObjectType) => {
     const { firstName, lastName, netid, phoneNumber, startDate, availability } =
       data;
-    if (selectedRole === 'admin') {
-      const admin = {
-        firstName,
-        lastName,
-        email: netid + '@cornell.edu',
-        phoneNumber,
-      };
-      if (existingEmployee) {
-        updateExistingEmployee(
-          admin,
-          '/api/admins',
-          () => refreshAdmins(),
-          'Admins'
-        );
+
+    const driver = {
+      firstName,
+      lastName,
+      email: netid + '@cornell.edu',
+      phoneNumber,
+      startDate,
+      availability: parseAvailability(availability),
+    };
+
+    const admin = {
+      firstName,
+      lastName,
+      email: netid + '@cornell.edu',
+      type: selectedRole.filter((role) => !(role === 'driver')),
+      phoneNumber,
+      availability: parseAvailability(availability),
+      isDriver: selectedRole.includes('driver'),
+    };
+
+    const existingDriver = existingEmployee?.isDriver === undefined;
+    const existingAdmin = existingEmployee?.isDriver !== undefined;
+
+    if (existingEmployee) {
+      if (selectedRole.includes('driver')) {
+        if (selectedRole.some((role) => role.includes('admin'))) {
+          if (existingDriver && existingAdmin) {
```

Comment:
I think this `if` chain would be more readable if the driver/admin logic is separated

also, either operations don't need to be sequential, so you can use Promise.all()

Reply:
because of bandwidth limitations, let's tackle this next sem